Gems/EMotionFX/Code/EMotionFX/Source/Recorder.h

```diff
@@ -193,50 +193,66 @@ namespace EMotionFX
 
         struct EMFX_API AnimGraphAnimObjectInfo
         {
-            uint32              mFrameByteOffset;
+            size_t              mFrameByteOffset;
             AnimGraphObject*   mObject;
         };
 
 
         struct EMFX_API AnimGraphAnimFrame
         {
-            float                                       mTimeValue;
-            uint32                                      mByteOffset;
-            uint32                                      mNumBytes;
-            MCore::Array<AnimGraphAnimObjectInfo>      mObjectInfos;
-            MCore::Array<MCore::Attribute*>             mParameterValues;
-
-            AnimGraphAnimFrame()
-            {
-                mTimeValue  = 0.0f;
-                mByteOffset = 0;
-                mNumBytes   = 0;
-            }
+            float                                       mTimeValue = 0.0f;
+            size_t                                      mByteOffset = 0;
+            size_t                                      mNumBytes = 0;
+            AZStd::vector<AnimGraphAnimObjectInfo>      mObjectInfos{};
+            AZStd::vector<AZStd::unique_ptr<MCore::Attribute>>             mParameterValues{};
```

Comment:
Did you test the recorder? We're cloning the attributes when recording and it should be safe, but let's give that a testing round as the recorder usually is pretty sensitive.

Reply:
I did some basic tests with the recorder and it seems to be working.